src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -986,6 +1016,88 @@ public override BoundNode VisitIsPatternExpression(BoundIsPatternExpression node
             }
 
             return node;
+
+            // Returns `true` if the pattern only matches if the top-level input is not null.
+            // Returns `false` if the pattern only matches if the top-level input is null.
+            // Otherwise, returns `null`.
+            static bool? isTopLevelNullTest(BoundPattern pattern)
+            {
+                switch (pattern)
+                {
+                    case BoundTypePattern:
```

Comment:
I'm not sure when `expr is T` results in a `BoundIsOperator` as opposed to the new pattern representation (`BoundIsPatternExpression`). We may need to handle that case too.

Reply:
Ah, I see that you handled that already (`VisitIsOperator` below). Never mind :-)

---
In reply to: [616348015](https://github.com/dotnet/roslyn/pull/52616#discussion_r616348015) [](ancestors = 616348015)